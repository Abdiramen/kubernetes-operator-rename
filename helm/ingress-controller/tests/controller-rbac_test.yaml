suite: test controller-rbac
templates:
- controller-rbac.yaml
tests:
- it: Should generate rbac
  asserts:
  - matchSnapshot: {}
