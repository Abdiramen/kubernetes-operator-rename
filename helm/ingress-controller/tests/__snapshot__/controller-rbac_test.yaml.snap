Should match snapshot:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      name: ngrok-ingress-controller-manager
      namespace: NAMESPACE
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      name: ngrok-ingress-controller-leader-election-role
      namespace: NAMESPACE
    rules:
    - apiGroups:
      - ""
      resources:
      - configmaps
      verbs:
      - get
      - list
      - watch
      - create
      - update
      - patch
      - delete
    - apiGroups:
      - coordination.k8s.io
      resources:
      - leases
      verbs:
      - get
      - list
      - watch
      - create
      - update
      - patch
      - delete
    - apiGroups:
      - ""
      resources:
      - events
      verbs:
      - create
      - patch
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: ngrok-ingress-controller-proxy-role
    rules:
    - apiGroups:
      - authentication.k8s.io
      resources:
      - tokenreviews
      verbs:
      - create
    - apiGroups:
      - authorization.k8s.io
      resources:
      - subjectaccessreviews
      verbs:
      - create
  4: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: ngrok-ingress-controller-leader-election-rolebinding
      namespace: NAMESPACE
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: ngrok-ingress-controller-leader-election-role
    subjects:
    - kind: ServiceAccount
      name: ngrok-ingress-controller-manager
      namespace: NAMESPACE
  5: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: ngrok-ingress-controller-manager-rolebinding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: ngrok-ingress-controller-manager-role
    subjects:
    - kind: ServiceAccount
      name: ngrok-ingress-controller-manager
      namespace: NAMESPACE
  6: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: ngrok-ingress-controller-proxy-rolebinding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: ngrok-ingress-controller-proxy-role
    subjects:
    - kind: ServiceAccount
      name: ngrok-ingress-controller-manager
      namespace: NAMESPACE
